assets/src/components/PostMeta/withPostMeta.js

```diff
@@ -14,13 +14,38 @@ const getValuePropName = ( Component ) => {
 
 const { apiFetch } = wp;
 
+const getValueFromProps = ( props ) => {
+  const { metaKey, postMeta } = props;
+
+  const metaValue = postMeta[ metaKey ];
+
+  let value;
+  // Use the default value if what is stored in the post meta is empty, or if it isn't one of the listed options.
+  if (
+    props.defaultValue
+    && (
+      !metaValue
+      || !props.options.some( option => option.value === metaValue )
+    )
+  ) {
+    value = props.defaultValue;
+  } else {
+    value = metaValue;
+  }
+
+  return value;
+}
+
 export function withPostMeta( WrappedComponent ) {
 
   class WrappingComponent extends Component {
     constructor( props ) {
       super( props );
       this.handleChange = this.handleChange.bind( this );
       this.valuePropName = getValuePropName( WrappedComponent );
+      this.state = {
+        value: getValueFromProps( props )
+      };
     }
 
     handleChange( metaKey, value ) {
```

Comment:
We can actually adapt the code below that updates the preview meta to use the function I introduced to do the same thing on page load: https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/0c211c994b7626a37b5c8b3f886181e24791a84a/assets/src/saveMetaToPreview.js

But maybe it actually would make more sense if we set this up as a listener to the `wp/data` store in some way? Then we wouldn't even need to call `savePreviewMeta` more than once. Though a challenge with that approach is that you need to listen to any event in the `wp/data` store and then only fire if the meta has changed. Or maybe there is a way after all to just listen post meta changes.

Reply:
It would also be an improvement over the current call to `savePreviewMeta` happening on the `DOMContentLoaded` event.